### PR TITLE
fix: add royalty fee suite tests

### LINKED ISSUE #11833: Add additional suites for `ROYALTY_FEE`

### PR DESCRIPTION
**Pull request by dikel**
**Description**:
This PR adds several suite test cases for custom `ROYALTY_FEE` in `TransferWithCustomFees`

**Related issue(s)**:

Fixes #11833

### CODE REVIEW

File: hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/crypto/TransferWithCustomRoyaltyFees.java [hunk 1]
```diff
@@ -0,0 +1,426 @@
...
+                        mintToken(nonFungibleToken, List.of(ByteStringUtils.wrapUnsafely("meta1".getBytes()))),
+                        cryptoTransfer(movingUnique(nonFungibleToken, 1L).between(tokenTreasury, tokenOwner)))
+                .when(cryptoTransfer(
+                        movingUnique(nonFungibleToken, 1L).between(tokenOwner, tokenReceiver),
+                        movingHbar(fourHundredHbars).between(tokenReceiver, tokenOwner)))
+                .then(
+                        getAccountBalance(tokenOwner)
+                                .hasTinyBars(twoHundredHbars)
+                                .hasTokenBalance(nonFungibleToken, 0),
+                        getAccountBalance(tokenReceiver).hasTokenBalance(nonFungibleToken, 1),
```
Comment (IvanKavaldzhiev) [inline hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/crypto/TransferWithCustomRoyaltyFees.java:132]: We can also validate that `tokenReceiver` has 400 hbars less compared to the balance before the `cryptoTransfer`.

File: hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/crypto/TransferWithCustomRoyaltyFees.java [hunk 2]
```diff
@@ -0,0 +1,426 @@
...
+                                        1, 2, fixedHtsFeeInheritingRoyaltyCollector(5, feeDenom), htsCollector)),
+                        tokenAssociate(tokenReceiver, nonFungibleToken),
+                        tokenAssociate(tokenOwner, nonFungibleToken),
+                        mintToken(nonFungibleToken, List.of(ByteStringUtils.wrapUnsafely("meta1".getBytes()))),
+                        cryptoTransfer(movingUnique(nonFungibleToken, 1L).between(tokenTreasury, tokenOwner)))
+                .when()
+                .then(
+                        cryptoTransfer(movingUnique(nonFungibleToken, 1L).between(tokenOwner, tokenReceiver))
+                                .signedByPayerAnd(tokenReceiver, tokenOwner)
+                                .hasKnownStatus(INSUFFICIENT_SENDER_ACCOUNT_BALANCE_FOR_CUSTOM_FEE),
```
Comment (IvanKavaldzhiev) [inline hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/crypto/TransferWithCustomRoyaltyFees.java:356]: Shouldn't we have the same error as for the hbar fallback fee - `INSUFFICIENT_ACCOUNT_BALANCE`? Have you found where the difference comes from?

**Review by petreze — APPROVED** [review]

**Review by stefan-stefanooov — APPROVED** [review]

**Review by petreze — APPROVED** [review]
LGTM!
Nice job!

**Review by tinker-michaelj — APPROVED** [review]
LGTM, tyvm @dikel !

**Pull request merged**